src/serac/numerics/solver_config.hpp

```diff
@@ -396,6 +396,16 @@ struct NonlinearSolverOptions {
   /// Scaling for the initial trust region size
   double trust_region_scaling = 0.1;
 
+  /// Specify subspace option
+  /// 0: no subspace solve
+  /// 1: subspace solve only when indefinite system found
+  /// 2: subspace solve only when indefinite system or step to boundary
+  /// 3: subspace solve always
+  int subspace_option = 0;
```

Comment:
would it be worth making this a enum?

Reply:
I think I like that.  Self documenting, less comments.